bin/git-sed

```diff
@@ -64,6 +66,22 @@ actual command:
     shift
 done
 
-command="git grep -lz '$search' | xargs -0 sed -i 's/$search/$replacement/$flags'"
-git grep -lz "$search" | xargs -0 sed -i "s/$search/$replacement/$flags"
+all="$search$replacement$flags"
+case "$all" in
+    */*)
+        ascii="$(for((i=32;i<=127;i++)) do printf '%b' "\\$(printf '%03o' "$i")"; done)"
+        sep="$(printf '%s' "$ascii" | tr -d "$all")"
+        sep="$(printf %.1s "$sep")"
+        if [ "X$sep" = "X" ] ; then
+            echo 'could not find an unused character for sed separator character'
+            exit 1
+        fi
+        ;;
+    *)
+        sep=/
+        ;;
+esac
+r=$(xargs -0r false < /dev/null > /dev/null 2>&1 && echo r)
```

Comment:
Maybe we could use `-r` instead of `-0r` here, since we only want to test for `-r`?